### ISSUE TITLE
UnboundLocalError: local variable 'depth_gt' referenced before assignment

### ISSUE THREAD

**Issue by Draksis-AleX** [maturk/dn-splatter#72]
Hi, I am trying to use the dn-splatter model on a dataset with colmap.
I get this error:

```
Traceback (most recent call last):
  File "/home/aprivitera/anaconda/envs/nerfstudio/bin/ns-train", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/scripts/train.py", line 262, in entrypoint
    main(
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/scripts/train.py", line 247, in main
    launch(
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/scripts/train.py", line 189, in launch
    main_func(local_rank=0, world_size=world_size, config=config)
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/scripts/train.py", line 100, in train_loop
    trainer.train()
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/engine/trainer.py", line 261, in train
    loss, loss_dict, metrics_dict = self.train_iteration(step)
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/utils/profiler.py", line 112, in inner
    out = func(*args, **kwargs)
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/engine/trainer.py", line 496, in train_iteration
    _, loss_dict, metrics_dict = self.pipeline.get_train_loss_dict(step=step)
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/utils/profiler.py", line 112, in inner
    out = func(*args, **kwargs)
  File "/home/aprivitera/anaconda/envs/nerfstudio/lib/python3.8/site-packages/nerfstudio/pipelines/base_pipeline.py", line 303, in get_train_loss_dict
    loss_dict = self.model.get_loss_dict(model_outputs, batch, metrics_dict)
  File "/home/aprivitera/dn-splatter/dn_splatter/dn_model.py", line 692, in get_loss_dict
    gt_depth=depth_gt,
UnboundLocalError: local variable 'depth_gt' referenced before assignment
```

This is the command I am using:
`ns-train dn-splatter --pipeline.model.use-depth-loss True --pipeline.model.depth-lambda 0.2 --pipeline.model.use-normal-loss True --pipeline.model.use-normal-tv-loss True --pipeline.model.normal-supervision mono --save-only-latest-checkpoint True --max-num-iterations 30000 --pipeline.model.use_scale_regularization True --pipeline.model.cull-alpha-thresh 0.005 --pipeline.model.continue_cull_post_densification=False --data {MY_DATAPATH} coolermap --load_normals True`

Before running the command I downloaded the omnimodel weights with:
`python dn_splatter/data/download_scripts/download_omnidata.py`

I also estimated the normals with:
`python dn_splatter/scripts/normals_from_pretrain.py --data-dir {MY_DATAPATH}`

**Comment by ferryman123** [comment]
@Draksis-AleX  I met the same problem with you, and I check the code  find that it doesn't have the definition about "self.get_gt_img". Have you solve this problem?

**Comment by ferryman123** [comment]
@maturk  Could you please answer the question above? Thank you.

**Comment by maturk** [comment]
@ferryman123 sorry for the delay, I have been pretty busy thesedays. So the problem is that by default, the coolermap dataparser has `--load-depths` set to False, which skips loading depths and the crash occurs since the dn_model is not able to find any depths for supervision when you set the `--pipeline.model.use-depth-loss True` flag.

I find it quite hard which settings should be set to true or false based on users needs. But I have now changed it so that --load-depths is always True for colmap data, and the program will just inform you that it could not find any depths if you did not compute them before hand.

#74 

If you have colmap data, first run:
`python dn_splatter/scripts/depth_from_pretrain.py --data-dir path_to_data` or `python dn_splatter/scripts/align_depth.py --data path_to_data` to generate zoe depth estimates or scale-aligned zoe depth estimates. 

use the --help command to see all options for the above scripts. 

When training, use the `--pipeline.model.depth-loss-type PearsonDepth` option. We found this to be the best depth loss for monocular supervision.

**Comment by ferryman123** [comment]
@maturk I tried generating all the required normal and depth information and then followed your steps, but it still doesn't work, still the same error.Here is the structure of my dataset
![截图 2024-09-26 16-31-48](https://github.com/user-attachments/assets/992afa17-6192-4a22-a5b4-4abfb4699281)

**Comment by ferryman123** [comment]
The command I used :
ns-train dn-splatter --pipeline.model.use-depth-loss True --pipeline.model.depth-loss-type PearsonDepth --pipeline.model.depth-lambda 0.2 --pipeline.model.use-normal-loss True --pipeline.model.use-normal-tv-loss True --pipeline.model.normal-supervision mono coolermap --data ./datasets/drjohnson/

**Comment by ferryman123** [comment]
@maturk The command I used :
ns-train dn-splatter --pipeline.model.use-depth-loss True --pipeline.model.depth-loss-type PearsonDepth --pipeline.model.depth-lambda 0.2 --pipeline.model.use-normal-loss True --pipeline.model.use-normal-tv-loss True --pipeline.model.normal-supervision mono coolermap --data ./datasets/drjohnson/

**Comment by maturk** [comment]
@ferryman123 , Did you git pull the latest from the main branch?

**Comment by ferryman123** [comment]
@maturk  Sorry, I didn't see to change the configuration of the coolermap_dataparser.py file at first, I changed them all to True,and it worked. Thank you very much for your answer!

**Comment by maturk** [comment]
Yeah you can git pull the latest main, I fixed some issues in #74 and added some warnings that will tell you if some setting is wrongly set.
